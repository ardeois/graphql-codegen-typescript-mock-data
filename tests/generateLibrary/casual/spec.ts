import { anA, seedMocks } from './mocks';

it('should generate dynamic values when dynamicValues is true', () => {
    const a1 = anA();
    expect(a1).toMatchSnapshot();

    const a2 = anA();
    expect(a2).toMatchSnapshot();

    expect(a1).not.toEqual(a2);
});

it('should generate dynamic values from seed when dynamicValues is true', () => {
    seedMocks(0);
    const a1 = anA();

    seedMocks(0);
    const a1Copy = anA();

    seedMocks(1);
    const a2 = anA();

    expect(a1).toEqual(a1Copy);
    expect(a1).not.toEqual(a2);
    expect(a1Copy).not.toEqual(a2);
});
