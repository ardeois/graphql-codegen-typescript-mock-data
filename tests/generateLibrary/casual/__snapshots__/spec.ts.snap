// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate dynamic values when dynamicValues is true 1`] = `
Object {
  "id": "99bd9d8d-79a6-474e-8f46-6cc8796ed151",
  "obj": Object {
    "bool": false,
    "flt": 3.68,
    "int": 202,
  },
  "str": "est",
}
`;

exports[`should generate dynamic values when dynamicValues is true 2`] = `
Object {
  "id": "ec2ddf7c-d78c-4a1b-a928-ec816742cb73",
  "obj": Object {
    "bool": true,
    "flt": 0.19,
    "int": 1352,
  },
  "str": "similique",
}
`;
